*** arpwatch-2.1a15/arpwatch.8	2007-07-11 21:50:28.000000000 +0200
--- arpwatch-2.1a15_ossim/arpwatch.8	2007-07-11 21:50:39.000000000 +0200
***************
*** 27,33 ****
  .na
  .B arpwatch
  [
! .B -dN
  ] [
  .B -f
  .I datafile
--- 27,33 ----
  .na
  .B arpwatch
  [
! .B -dNw
  ] [
  .B -f
  .I datafile
***************
*** 83,88 ****
--- 83,92 ----
  flag disables reporting any bogons.
  .LP
  The
+ .B -w
+ flag is used to specify an interface without a valid IPv4 address.
+ .LP
+ The
  .B -r
  flag is used to specify a savefile
  (perhaps created by
