*** arpwatch-2.1a15/arpwatch.c	2007-07-11 20:34:36.000000000 +0200
--- arpwatch-2.1a15_ossim/arpwatch.c	2007-07-11 20:51:06.000000000 +0200
***************
*** 145,151 ****
  main(int argc, char **argv)
  {
  	register char *cp;
! 	register int op, pid, snaplen, timeout, linktype, status;
  #ifdef TIOCNOTTY
  	register int fd;
  #endif
--- 145,151 ----
  main(int argc, char **argv)
  {
  	register char *cp;
! 	register int op, pid, snaplen, timeout, linktype, status, noip=0;
  #ifdef TIOCNOTTY
  	register int fd;
  #endif
***************
*** 170,176 ****
  	interface = NULL;
  	rfilename = NULL;
  	pd = NULL;
! 	while ((op = getopt(argc, argv, "df:i:n:Nr:")) != EOF)
  		switch (op) {
  
  		case 'd':
--- 170,176 ----
  	interface = NULL;
  	rfilename = NULL;
  	pd = NULL;
! 	while ((op = getopt(argc, argv, "df:i:n:wNr:")) != EOF)
  		switch (op) {
  
  		case 'd':
***************
*** 201,206 ****
--- 201,209 ----
  		case 'r':
  			rfilename = optarg;
  			break;
+ 		case 'w':
+ 			++noip;
+ 			break;
  
  		default:
  			usage();
***************
*** 221,232 ****
--- 224,241 ----
  			exit(1);
  		}
  
+ 		if(noip){
+ 		/* Be able to sniff on interfaces without address */
+ 			net = 0;
+ 			netmask = 0;
+ 		} else {
  		/* Determine network and netmask */
  		if (pcap_lookupnet(interface, &net, &netmask, errbuf) < 0) {
  			(void)fprintf(stderr, "%s: bad interface %s: %s\n",
  			    prog, interface, errbuf);
  			exit(1);
  		}
+ 		}
  
  		/* Drop into the background if not debugging */
  		if (!debug) {
***************
*** 750,756 ****
  	extern char version[];
  
  	(void)fprintf(stderr, "Version %s\n", version);
! 	(void)fprintf(stderr, "usage: %s [-dN] [-f datafile] [-i interface]"
  	    " [-n net[/width]] [-r file]\n", prog);
  	exit(1);
  }
--- 759,765 ----
  	extern char version[];
  
  	(void)fprintf(stderr, "Version %s\n", version);
! 	(void)fprintf(stderr, "usage: %s [-dNw] [-f datafile] [-i interface]"
  	    " [-n net[/width]] [-r file]\n", prog);
  	exit(1);
  }
