diff -aur phpgacl-3.3.7.orig/debian/apache.conf phpgacl-3.3.7/debian/apache.conf
--- phpgacl-3.3.7.orig/debian/apache.conf	2007-08-11 19:13:28.000000000 +0200
+++ phpgacl-3.3.7/debian/apache.conf	2007-08-11 19:14:36.000000000 +0200
@@ -3,9 +3,9 @@
 </IfModule>
 
 <Location /phpgacl>
-    order deny,allow
-    deny from all
-    allow from 127.0.0.0/255.0.0.0
+#    order deny,allow
+#    deny from all
+#    allow from 127.0.0.0/255.0.0.0
     Options Indexes FollowSymLinks MultiViews
 </Location>
 
diff -aur phpgacl-3.3.7.orig/debian/phpgacl.config phpgacl-3.3.7/debian/phpgacl.config
--- phpgacl-3.3.7.orig/debian/phpgacl.config	2007-08-11 19:13:28.000000000 +0200
+++ phpgacl-3.3.7/debian/phpgacl.config	2007-08-11 19:15:35.000000000 +0200
@@ -5,6 +5,8 @@
 
 # source dbconfig-common stuff
 dbc_dbtypes="mysql, pgsql"
+dbc_dbuser="root"
+dbc_dbname="ossim_acl"
 
 if [ -f /usr/share/dbconfig-common/dpkg/config ]; then
     . /usr/share/dbconfig-common/dpkg/config
