diff -urNb ntop/Makefile.am ntop-3.1-ossim/Makefile.am
--- ntop/Makefile.am	Tue Dec 21 00:22:40 2004
+++ ntop-3.1-ossim/Makefile.am	Wed Jan  5 11:35:43 2005
@@ -33,7 +33,9 @@
 DEFS      = @DEFS@
 INCS      = @INCS@
 
-INCLUDES  = -I. $(INCS) -Imyrrd
+MYRRD_INC = /usr/local/rrdtool-1.1.0/include
+
+INCLUDES  = -I. $(INCS) -I$(MYRRD_INC)
 LIBS      = @LIBS@
 AS        = @AS@
 AWK       = @AWK@
@@ -61,7 +63,7 @@
                        stamp-h.in
 
 
-SUBDIRS = myrrd . @PLUGINS@
+SUBDIRS = @PLUGINS@
 
 ETTER_PASSIVE_FILE = etter.finger.os
 ETTER_PASSIVE = $(ETTER_PASSIVE_FILE).gz
@@ -102,8 +104,8 @@
 bin_SCRIPTS    = 
 
 ntop_SOURCES       = main.c admin.c
-ntop_DEPENDENCIES  = libntopreport.la libntop.la myrrd/libmyrrd.la
-ntop_LDADD         = libntopreport.la libntop.la myrrd/libmyrrd.la
+ntop_DEPENDENCIES  = libntopreport.la libntop.la
+ntop_LDADD         = libntopreport.la libntop.la
 ntop_LDFLAGS       = $(AM_LDFLAGS)
 ntop_CFLAGS        = 
 
@@ -137,7 +139,7 @@
                      traffic.c    vendor.c      version.c      \
                      ntop_darwin.c
 
-libntop_la_DEPENDENCIES =  myrrd/libmyrrd.la config.h
+libntop_la_DEPENDENCIES =  config.h
 libntop_la_LIBADD       = 
 libntop_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
 
@@ -148,14 +150,11 @@
                            webInterface.c
 
 
-libntopreport_la_DEPENDENCIES =  myrrd/libmyrrd.la config.h
+libntopreport_la_DEPENDENCIES =  config.h
 libntopreport_la_LIBADD       = 
 libntopreport_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
 
 man_MANS = ntop.8
-
-myrrd/libmyrrd.la:
-	cd myrrd; make
 
 .PHONY: snapshot
 
diff -urNb ntop/configure.in ntop-3.1-ossim/configure.in
--- ntop/configure.in	Tue Dec 21 05:07:08 2004
+++ ntop-3.1-ossim/configure.in	Wed Jan  5 11:27:57 2005
@@ -128,7 +128,7 @@
 AC_PREFIX_DEFAULT(${NTOPDIR:-/usr/local})
 
 # Set constants and defaults
-MYRRD=`pwd`/myrrd
+MYRRD=/usr/local/rrdtool-1.1.0/lib
 SO_VERSION_PATCH=
 MAKE_SHARED_LIBRARY_PARM="-bundle -flat_namespace -undefined suppress"
 
@@ -2848,7 +2848,7 @@
 fi
 
 dnl> Append RRD finally
-dnl> NTOP_APPENDS([${MYRRD}], [${MYRRD}], [myrrd])
+NTOP_APPENDS([${MYRRD}], [${MYRRD}], [rrd_th])
 
 echo
 echo "-------------------------------------------------------------------"
@@ -3024,7 +3024,6 @@
 AC_SUBST(SO_VERSION_PATCH)
 
 AC_CONFIG_FILES([Makefile])
-AC_CONFIG_FILES([myrrd/Makefile])
 
 if test ".${ac_disable_plugins}" != ".yes"; then
     AC_CONFIG_FILES([plugins/Makefile])
diff -urNb ntop/globals-defines.h ntop-3.1-ossim/globals-defines.h
--- ntop/globals-defines.h	Mon Dec 20 16:34:33 2004
+++ ntop-3.1-ossim/globals-defines.h	Wed Jan  5 10:46:36 2005
@@ -569,7 +569,7 @@
  *
  * Valid values are 0 (disables) up to whatever.
  */
-#define MAX_NUM_BAD_IP_ADDRESSES            3
+#define MAX_NUM_BAD_IP_ADDRESSES            0
 
 
 /* Maximum number of queued addresses waiting to be resolved */
diff -urNb ntop/globals-structtypes.h ntop-3.1-ossim/globals-structtypes.h
--- ntop/globals-structtypes.h	Mon Dec 20 16:34:33 2004
+++ ntop-3.1-ossim/globals-structtypes.h	Wed Jan  5 10:46:41 2005
@@ -461,6 +461,8 @@
   UsageCounter icmpProtocolUnreachSent, icmpProtocolUnreachRcvd;
   UsageCounter icmpAdminProhibitedSent, icmpAdminProhibitedRcvd;
   UsageCounter malformedPktsSent,       malformedPktsRcvd;
+  /* OSSIM */
+  UsageCounter mailSynPkts, webSynPkts, nbSynPkts;
 } SecurityHostProbes;
 
 /* NOTE: anything added here must be also added in the SecurityHostProbes structure */
diff -urNb ntop/plugins/Makefile.am ntop-3.1-ossim/plugins/Makefile.am
--- ntop/plugins/Makefile.am	Thu Nov 11 11:12:52 2004
+++ ntop-3.1-ossim/plugins/Makefile.am	Wed Jan  5 11:21:03 2005
@@ -37,7 +37,9 @@
 # 
 plugindir = $(libdir)/ntop/plugins
 
-INCLUDES = -I.. @INCS@ -I../myrrd
+MYRRD_INC = /usr/local/rrdtool-1.1.0/include
+
+INCLUDES = -DHAVE_RRD_ABERRANT_BEHAVIOR -I.. -I$(MYRRD_INC) @INCS@
 
 #
 # The meat for ntop
@@ -76,7 +78,7 @@
 
 librrdPlugin_la_SOURCES = rrdPlugin.c
 librrdPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
-librrdPlugin_la_LIBADD = -L../myrrd/ -lmyrrd
+librrdPlugin_la_LIBADD = -lrrd_th
 
 libsnmpPlugin_la_SOURCES = snmpPlugin.c
 libsnmpPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
diff -urNb ntop/plugins/rrdPlugin.c ntop-3.1-ossim/plugins/rrdPlugin.c
--- ntop/plugins/rrdPlugin.c	Wed Dec 15 22:50:31 2004
+++ ntop-3.1-ossim/plugins/rrdPlugin.c	Wed Jan  5 15:00:11 2005
@@ -501,7 +501,7 @@
 		  char *startTime, char* endTime, char *rrdPrefix) {
   char path[512], *argv[32], buf[384], buf1[384], fname[384], *label, tmpStr[32];
 #ifdef HAVE_RRD_ABERRANT_BEHAVIOR
-  char buf2[384], buf3[384], buf4[384];
+  char buf2[384], buf3[384], buf4[384], buf5[384];
 #endif
   struct stat statbuf;
   int argc = 0, rc, x, y;
@@ -564,7 +564,8 @@
     argv[argc++] = buf3;
     safe_snprintf(__FILE__, __LINE__, buf4, sizeof(buf4), "DEF:fail=%s:counter:FAILURES", path);
     argv[argc++] = buf4;
-    argv[argc++] = "TICK:fail#ffffa0:1.0:Anomalia";
+    safe_snprintf(__FILE__, __LINE__, buf5, sizeof(buf5), "TICK:fail#ffffa0:%.1f:Anomaly", 0.2);
+    argv[argc++] = buf5;
     argv[argc++] = "CDEF:upper=pred,dev,2,*,+";
     argv[argc++] = "CDEF:lower=pred,dev,2,*,-";
     argv[argc++] = "LINE1:upper#ff0000:Upper";
@@ -580,7 +581,11 @@
     fillupArgv(argc, sizeof(argv)/sizeof(char*), argv);
     rrd_clear_error();
     addRrdDelay();
+#ifdef HAVE_RRD_ABERRANT_BEHAVIOR
+    rc = rrd_graph(argc, argv, &calcpr, &x, &y, NULL);
+#else
     rc = rrd_graph(argc, argv, &calcpr, &x, &y);
+#endif //HAVE_RRD_ABERRANT_BEHAVIOR
 
     calfree();
 
@@ -725,7 +730,11 @@
   fillupArgv(argc, sizeof(argv)/sizeof(char*), argv);
   rrd_clear_error();
   addRrdDelay();
+#ifdef HAVE_RRD_ABERRANT_BEHAVIOR
+    rc = rrd_graph(argc, argv, &calcpr, &x, &y, NULL);
+#else
   rc = rrd_graph(argc, argv, &calcpr, &x, &y);
+#endif //HAVE_RRD_ABERRANT_BEHAVIOR
 
   calfree();
 
@@ -923,7 +932,11 @@
   fillupArgv(argc, sizeof(argv)/sizeof(char*), argv);
   rrd_clear_error();
   addRrdDelay();
+#ifdef HAVE_RRD_ABERRANT_BEHAVIOR
+    rc = rrd_graph(argc, argv, &calcpr, &x, &y, NULL);
+#else
   rc = rrd_graph(argc, argv, &calcpr, &x, &y);
+#endif //HAVE_RRD_ABERRANT_BEHAVIOR
 
   calfree();
 
@@ -1042,7 +1055,7 @@
     }
 
 #ifdef HAVE_RRD_ABERRANT_BEHAVIOR
-    safe_snprintf(__FILE__, __LINE__, tempStr, sizeof(tempStr), "RRA:HWPREDICT:1440:0.1:0.0035:20");
+    safe_snprintf(__FILE__, __LINE__, tempStr, sizeof(tempStr), "RRA:HWPREDICT:1440:%.1f:%.4f:20", 0.1, 0.0035);
     argv[argc++] = tempStr;
 #endif
 
@@ -2036,6 +2049,14 @@
 	updateTrafficCounter(rrdPath, "netbiosRcvd", &el->nonIPTraffic->netbiosRcvd);
 	updateTrafficCounter(rrdPath, "otherSent", &el->nonIPTraffic->otherSent);
 	updateTrafficCounter(rrdPath, "otherRcvd", &el->nonIPTraffic->otherRcvd);
+      }
+
+      if(el->secHostPkts) {
+    updateCounter(rrdPath, "synPktsSent", el->secHostPkts->synPktsSent.value.value);
+    updateCounter(rrdPath, "synPktsRcvd", el->secHostPkts->synPktsRcvd.value.value);
+    updateCounter(rrdPath, "mail_sessions", el->secHostPkts->mailSynPkts.value.value);
+    updateCounter(rrdPath, "web_sessions", el->secHostPkts->webSynPkts.value.value);
+    updateCounter(rrdPath, "nb_sessions", el->secHostPkts->nbSynPkts.value.value);
       }
 
       protoList = myGlobals.ipProtosList, idx=0;
diff -urNb ntop/sessions.c ntop-3.1-ossim/sessions.c
--- ntop/sessions.c	Thu Dec  2 18:12:01 2004
+++ ntop-3.1-ossim/sessions.c	Wed Jan  5 10:46:54 2005
@@ -1425,6 +1425,26 @@
       incrementTrafficCounter(&myGlobals.device[actualDeviceId].securityPkts.synPkts, 1);
     }
 
+    switch (dport)
+    {
+    case 25:
+        incrementUsageCounter(&srcHost->secHostPkts->mailSynPkts, dstHost, actualDeviceId);
+        incrementUsageCounter(&dstHost->secHostPkts->mailSynPkts, srcHost, actualDeviceId);
+        break;
+    case 80:
+        incrementUsageCounter(&srcHost->secHostPkts->webSynPkts, dstHost, actualDeviceId);
+        incrementUsageCounter(&dstHost->secHostPkts->webSynPkts, srcHost, actualDeviceId);
+        break;
+    case 135:
+    case 137:
+    case 138:
+    case 139:
+    case 445:
+        incrementUsageCounter(&srcHost->secHostPkts->nbSynPkts, dstHost, actualDeviceId);
+        incrementUsageCounter(&dstHost->secHostPkts->nbSynPkts, srcHost, actualDeviceId);
+    break;
+    }
+
   }
 
   if(tp->th_flags == (TH_RST|TH_ACK)) {
