diff -r -c ntop-2.2.93/ntop/Makefile.in ntop-2.2.93-dk/ntop/Makefile.in
*** ntop-2.2.93/ntop/Makefile.in	Sun Aug 17 01:28:33 2003
--- ntop-2.2.93-dk/ntop/Makefile.in	Mon Sep 15 15:49:30 2003
***************
*** 150,156 ****
                         stamp-h.in
  
  
! SUBDIRS = myrrd . @PLUGINS@
  
  ETTER_PASSIVE_DOWNLOAD = etter.passive.os.fp
  ETTER_PASSIVE = $(ETTER_PASSIVE_DOWNLOAD).gz
--- 150,156 ----
                         stamp-h.in
  
  
! SUBDIRS = @PLUGINS@
  
  ETTER_PASSIVE_DOWNLOAD = etter.passive.os.fp
  ETTER_PASSIVE = $(ETTER_PASSIVE_DOWNLOAD).gz
***************
*** 252,258 ****
                       ntop_darwin.c
  
  
! libntop_la_DEPENDENCIES = myrrd/libmyrrd.a config.h
  libntop_la_LIBADD = 
  libntop_la_LDFLAGS = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
  
--- 252,258 ----
                       ntop_darwin.c
  
  
! libntop_la_DEPENDENCIES = config.h
  libntop_la_LIBADD = 
  libntop_la_LDFLAGS = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
  
***************
*** 263,269 ****
                             webInterface.c  xmldump.c
  
  
! libntopreport_la_DEPENDENCIES = myrrd/libmyrrd.a config.h
  libntopreport_la_LIBADD = 
  libntopreport_la_LDFLAGS = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
  
--- 263,269 ----
                             webInterface.c  xmldump.c
  
  
! libntopreport_la_DEPENDENCIES = config.h
  libntopreport_la_LIBADD = 
  libntopreport_la_LDFLAGS = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
  
***************
*** 272,278 ****
  ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
  mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
  CONFIG_HEADER = config.h
- CONFIG_CLEAN_FILES = myrrd/Makefile
  LTLIBRARIES = $(lib_LTLIBRARIES)
  
  am_libntop_la_OBJECTS = address.lo argv.lo dataFormat.lo getopt.lo \
--- 272,277 ----
***************
*** 394,402 ****
  
  distclean-hdr:
  	-rm -f config.h
- myrrd/Makefile: $(top_builddir)/config.status $(top_srcdir)/myrrd/Makefile.in
- 	cd $(top_builddir) && $(SHELL) ./config.status $@
- libLTLIBRARIES_INSTALL = $(INSTALL)
  install-libLTLIBRARIES: $(lib_LTLIBRARIES)
  	@$(NORMAL_INSTALL)
  	$(mkinstalldirs) $(DESTDIR)$(libdir)
--- 393,398 ----
***************
*** 881,887 ****
  distdir: $(DISTFILES)
  	$(am__remove_distdir)
  	mkdir $(distdir)
- 	$(mkinstalldirs) $(distdir)/myrrd
  	@for file in $(DISTFILES); do \
  	  if test -f $$file || test -d $$file; then d=.; else d=$(srcdir); fi; \
  	  dir=`echo "$$file" | sed -e 's,/[^/]*$$,,'`; \
--- 877,882 ----
***************
*** 1071,1078 ****
  	uninstall-man8 uninstall-recursive
  
  
- myrrd/libmyrrd.a:
- 	cd myrrd; make
  
  #
  # Special stuff for xml auto-generation of code from the "doc":
--- 1066,1071 ----
diff -r -c ntop-2.2.93/ntop/configure ntop-2.2.93-dk/ntop/configure
*** ntop-2.2.93/ntop/configure	Fri Aug 22 19:36:19 2003
--- ntop-2.2.93-dk/ntop/configure	Mon Sep 15 14:28:55 2003
***************
*** 1933,1938 ****
--- 1933,1940 ----
  
  
  # Set constants and defaults
+ LIBRRD=/usr/local/rrdtool-1.1.0/lib
+ INCRRD=/usr/local/rrdtool-1.1.0/include
  MYRRD=`pwd`/myrrd
  SO_VERSION_PATCH=
  
***************
*** 24235,24245 ****
      if test ".${MYRRD}" != "."; then
          rc=`(echo $CPPFLAGS | grep '${MYRRD} ' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!            CPPFLAGS="$CPPFLAGS -I${MYRRD}"
          fi
          rc=`(echo $INCS | grep '${MYRRD}' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!             INCS="$INCS -I${MYRRD}"
          fi
      fi
      if test ".${MYRRD}" != "."; then
--- 24237,24247 ----
      if test ".${MYRRD}" != "."; then
          rc=`(echo $CPPFLAGS | grep '${MYRRD} ' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!            CPPFLAGS="$CPPFLAGS -I${INCRRD}"
          fi
          rc=`(echo $INCS | grep '${MYRRD}' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!             INCS="$INCS -I${INCRRD}"
          fi
      fi
      if test ".${MYRRD}" != "."; then
***************
*** 24247,24256 ****
          if [ $rc -eq 1 ]; then
              case "${DEFINEOS}" in
                DARWIN )
!                 LDFLAGS="$LDFLAGS -L${MYRRD} -L${MYRRD}/lib"
                  ;;
                * )
!                 LDFLAGS="$LDFLAGS -L${MYRRD}"
                  ;;
              esac
          fi
--- 24249,24258 ----
          if [ $rc -eq 1 ]; then
              case "${DEFINEOS}" in
                DARWIN )
!                 LDFLAGS="$LDFLAGS -L${LIBRRD} -L${LIBRRD}/lib"
                  ;;
                * )
!                 LDFLAGS="$LDFLAGS -L${LIBRRD}"
                  ;;
              esac
          fi
***************
*** 24258,24264 ****
      if test ".myrrd" != "."; then
          rc=`(echo $LIBS | grep '\-lmyrrd ' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!             LIBS="$LIBS -lmyrrd"
          fi
      fi
  # Finished expansion of NTOP_APPENDS()
--- 24260,24266 ----
      if test ".myrrd" != "."; then
          rc=`(echo $LIBS | grep '\-lmyrrd ' > /dev/null 2> /dev/null; echo $?)`
          if [ $rc -eq 1 ]; then
!             LIBS="$LIBS -lrrd -L${LIBRRD}"
          fi
      fi
  # Finished expansion of NTOP_APPENDS()
***************
*** 24292,24298 ****
  
            ac_config_files="$ac_config_files Makefile"
  
!           ac_config_files="$ac_config_files myrrd/Makefile"
  
  
  if test ".${ac_disable_plugins}" != ".yes"; then
--- 24294,24300 ----
  
            ac_config_files="$ac_config_files Makefile"
  
! #          ac_config_files="$ac_config_files myrrd/Makefile"
  
  
  if test ".${ac_disable_plugins}" != ".yes"; then
***************
*** 24842,24848 ****
    case "$ac_config_target" in
    # Handling of arguments.
    "Makefile" ) CONFIG_FILES="$CONFIG_FILES Makefile" ;;
!   "myrrd/Makefile" ) CONFIG_FILES="$CONFIG_FILES myrrd/Makefile" ;;
    "plugins/Makefile" ) CONFIG_FILES="$CONFIG_FILES plugins/Makefile" ;;
    "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
    "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
--- 24844,24850 ----
    case "$ac_config_target" in
    # Handling of arguments.
    "Makefile" ) CONFIG_FILES="$CONFIG_FILES Makefile" ;;
! #  "myrrd/Makefile" ) CONFIG_FILES="$CONFIG_FILES myrrd/Makefile" ;;
    "plugins/Makefile" ) CONFIG_FILES="$CONFIG_FILES plugins/Makefile" ;;
    "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
    "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
diff -r -c ntop-2.2.93/ntop/configure.in ntop-2.2.93-dk/ntop/configure.in
*** ntop-2.2.93/ntop/configure.in	Fri Aug 22 19:36:18 2003
--- ntop-2.2.93-dk/ntop/configure.in	Mon Sep 15 15:50:32 2003
***************
*** 128,134 ****
  AC_PREFIX_DEFAULT(${NTOPDIR:-/usr/local})
  
  # Set constants and defaults
! MYRRD=`pwd`/myrrd
  SO_VERSION_PATCH=
  
  AS=
--- 128,135 ----
  AC_PREFIX_DEFAULT(${NTOPDIR:-/usr/local})
  
  # Set constants and defaults
! LIBRRD=/usr/local/rrdtool-1.1.0/lib
! INCRRD=/usr/local/rrdtool-1.1.0/include
  SO_VERSION_PATCH=
  
  AS=
***************
*** 2844,2852 ****
  AC_DEFINE_UNQUOTED(CFG_DBFILE_DIR,"$CFG_DBFILE_DIR", [Database file directory])
  echo "    Database files are in $CFG_DBFILE_DIR"
  
- dnl> Append RRD finally
- NTOP_APPENDS([${MYRRD}], [${MYRRD}], [myrrd])
- 
  echo
  echo "-------------------------------------------------------------------"
  echo
--- 2845,2850 ----
***************
*** 2874,2880 ****
  AC_SUBST(SO_VERSION_PATCH)
  
  AC_CONFIG_FILES([Makefile])
- AC_CONFIG_FILES([myrrd/Makefile])
  
  if test ".${ac_disable_plugins}" != ".yes"; then
      AC_CONFIG_FILES([plugins/Makefile])
--- 2872,2877 ----
diff -r -c ntop-2.2.93/ntop/http.c ntop-2.2.93-dk/ntop/http.c
*** ntop-2.2.93/ntop/http.c	Fri Aug 22 19:30:58 2003
--- ntop-2.2.93-dk/ntop/http.c	Mon Sep 15 14:30:23 2003
***************
*** 2386,2391 ****
--- 2386,2392 ----
    int skipLeading, postLen, usedFork = 0;
    char requestedURL[MAX_LEN_URL], pw[64], agent[256];
    int rc, i;
+   int found = 0;
    struct timeval httpRequestedAt;
    u_int gzipBytesSent = 0;
  
***************
*** 2504,2512 ****
        code (why bother wasting the work effort
            Burton M. Strauss III <Burton@ntopsupport.com>, June 2002
      */
- 
-     int found = 0;
-     
      /* 
         Let's record the IP address of this nasty
         guy so he will stay far from ntop
--- 2505,2510 ----
Only in ntop-2.2.93/ntop: libtool
Only in ntop-2.2.93-dk/ntop/plugins: .deps
Only in ntop-2.2.93-dk/ntop/plugins: Makefile
diff -r -c ntop-2.2.93/ntop/plugins/Makefile.in ntop-2.2.93-dk/ntop/plugins/Makefile.in
*** ntop-2.2.93/ntop/plugins/Makefile.in	Wed Aug  6 17:55:47 2003
--- ntop-2.2.93-dk/ntop/plugins/Makefile.in	Mon Sep 15 15:50:04 2003
***************
*** 163,169 ****
  
  librrdPlugin_la_SOURCES = rrdPlugin.c
  librrdPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
! librrdPlugin_la_LIBADD = -L../myrrd -lmyrrd
  
  libpdaPlugin_la_SOURCES = pdaPlugin.c
  libpdaPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
--- 163,169 ----
  
  librrdPlugin_la_SOURCES = rrdPlugin.c
  librrdPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
! librrdPlugin_la_LIBADD = -L/usr/local/rrdtool-1.1.0/lib -lrrd
  
  libpdaPlugin_la_SOURCES = pdaPlugin.c
  libpdaPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
***************
*** 295,301 ****
  .SUFFIXES: .so .c .lo .o .obj
  $(srcdir)/Makefile.in:  Makefile.am  $(top_srcdir)/configure.in $(ACLOCAL_M4)
  	cd $(top_srcdir) && \
! 	  $(AUTOMAKE) --gnu  plugins/Makefile
  Makefile:  $(srcdir)/Makefile.in  $(top_builddir)/config.status
  	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
  libLTLIBRARIES_INSTALL = $(INSTALL)
--- 295,301 ----
  .SUFFIXES: .so .c .lo .o .obj
  $(srcdir)/Makefile.in:  Makefile.am  $(top_srcdir)/configure.in $(ACLOCAL_M4)
  	cd $(top_srcdir) && \
! 	   $(AUTOMAKE) --gnu  plugins/Makefile
  Makefile:  $(srcdir)/Makefile.in  $(top_builddir)/config.status
  	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
  libLTLIBRARIES_INSTALL = $(INSTALL)
diff -r -c ntop-2.2.93/ntop/plugins/rrdPlugin.c ntop-2.2.93-dk/ntop/plugins/rrdPlugin.c
*** ntop-2.2.93/ntop/plugins/rrdPlugin.c	Thu Aug 21 17:01:09 2003
--- ntop-2.2.93-dk/ntop/plugins/rrdPlugin.c	Mon Sep 15 14:22:23 2003
***************
*** 21,26 ****
--- 21,27 ----
  /* This plugin works only with threads */
  
  /* #define RRD_DEBUG  8 */
+ #define HAVE_RRD_ABERRANT_BEHAVIOR
  
  /*
  
Only in ntop-2.2.93-dk/ntop: trivialcompilemessage
Only in ntop-2.2.93-dk/ntop: version.c
Only in ntop-2.2.93-dk/ntop: xml_s_ntopinterface.inc
