*** rrdPlugin.c.orig	Fri Jul 18 14:58:56 2003
--- rrdPlugin.c	Fri Jul 18 14:58:29 2003
***************
*** 408,414 ****
  
  void graphCounter(char *rrdPath, char *rrdName, char *rrdTitle,
  		  char *startTime, char* endTime, char *rrdPrefix) {
!   char path[512], *argv[32], buf[384], buf1[384], fname[384], *label;
    struct stat statbuf;
    struct stat reusebuf;
    int argc = 0, rc, x, y;
--- 408,414 ----
  
  void graphCounter(char *rrdPath, char *rrdName, char *rrdTitle,
  		  char *startTime, char* endTime, char *rrdPrefix) {
!   char path[512], *argv[32], buf[384], buf1[384], buf2[384], buf3[384], buf4[384], fname[384], *label;
    struct stat statbuf;
    struct stat reusebuf;
    int argc = 0, rc, x, y;
***************
*** 486,491 ****
--- 486,507 ----
        argv[argc++] = "GPRINT:ctr:MAX:Max\\: %3.1lf%s";
        argv[argc++] = "GPRINT:ctr:AVERAGE:Avg\\: %3.1lf%s";
        argv[argc++] = "GPRINT:ctr:LAST:Current\\: %3.1lf%s";
+       snprintf(buf2,sizeof(buf2), "DEF:pred=%s:counter:HWPREDICT", path);
+       argv[argc++] = buf2;
+       snprintf(buf3,sizeof(buf3), "DEF:dev=%s:counter:DEVPREDICT", path);
+       argv[argc++] = buf3;
+       snprintf(buf4,sizeof(buf4), "DEF:fail=%s:counter:FAILURES", path);
+       argv[argc++] = buf4;
+       argv[argc++] = "TICK:fail#ffffa0:1.0:Anomaly";
+       argv[argc++] = "CDEF:upper=pred,dev,2,*,+";
+       argv[argc++] = "CDEF:lower=pred,dev,2,*,-";
+       argv[argc++] = "LINE1:upper#ff0000:Upper";
+       argv[argc++] = "LINE2:lower#ff0000:Lower";
+       /*
+       argv[argc++] = "--font TITLE:11:/tmp/arial.ttf";
+       argv[argc++] = "--font AXIS:5:/tmp/arial.ttf";
+       */
+ 
  
  #if RRD_DEBUG >= 3
        for (x = 0; x < argc; x++)
***************
*** 545,551 ****
  #endif
  
    if(stat(path, &statbuf) != 0) {
!     char startStr[32], stepStr[32], counterStr[64], intervalStr[32], minStr[32], maxStr[32], daysStr[32], monthsStr[32];
      int step = dumpInterval;
      int value1, value2;
      unsigned long topValue;
--- 561,567 ----
  #endif
  
    if(stat(path, &statbuf) != 0) {
!     char startStr[32], stepStr[32], counterStr[64], tempStr[64], intervalStr[32], minStr[32], maxStr[32], daysStr[32], monthsStr[32];
      int step = dumpInterval;
      int value1, value2;
      unsigned long topValue;
***************
*** 603,608 ****
--- 619,628 ----
          snprintf(monthsStr, sizeof(monthsStr), "RRA:AVERAGE:%.1f:%d:%d", 0.5, value1, dumpMonths * 30);
          argv[argc++] = monthsStr;
      }
+ 
+     /* Integramos Anomalias */
+     snprintf(tempStr, sizeof(tempStr), "RRA:HWPREDICT:1440:%.1f:%.4f:20", 0.1, 0.0035);
+     argv[argc++] = tempStr;
  
      if (shownCreate == 0) {
          char buf[LEN_GENERAL_WORK_BUFFER];
