diff -aur snort-2.7.0.orig/debian/rules snort-2.7.0/debian/rules
--- snort-2.7.0.orig/debian/rules	2007-11-05 11:57:42.000000000 +0100
+++ snort-2.7.0/debian/rules	2007-11-05 11:59:11.000000000 +0100
@@ -146,6 +146,7 @@
 	install -m 755 -o root -g root `pwd`/debian/snort.cron.daily $(TMP)/snort-common/etc/cron.daily/5snort
 	install -m 755 -o root -g root `pwd`/debian/my/snort-stat $(TMP)/snort-common/usr/sbin/
 	install -m 644 -o root -g root `pwd`/etc/snort.conf $(TMP)/snort-common/etc/snort/
+	install -m 644 -o root -g root `pwd`/etc/spade.ossim.conf $(TMP)/snort-common/etc/snort/
 	install -m 644 -o root -g root `pwd`/etc/unicode.map $(TMP)/snort-common/etc/snort/
 	install -m 644 -o root -g root `pwd`/etc/threshold.conf $(TMP)/snort-common/etc/snort/
 
diff -aur snort-2.7.0.orig/etc/snort.conf snort-2.7.0/etc/snort.conf
--- snort-2.7.0.orig/etc/snort.conf	2007-11-05 11:57:42.000000000 +0100
+++ snort-2.7.0/etc/snort.conf	2007-11-05 11:58:16.000000000 +0100
@@ -810,6 +810,8 @@
 # </debian>
 #
 
+### OSSIM output database ###
+#output database: alert, mysql, user=root password=ossim dbname=snort host=localhost sensor_name=replace_with_sensor_ip logfile=alert
 
 # unified: Snort unified binary format alerting and logging
 # -------------------------------------------------------------
@@ -829,6 +831,8 @@
 # output alert_unified: filename snort.alert, limit 128
 # output log_unified: filename snort.log, limit 128
 
+### OSSIM Unified format ###
+#output unified: filename snort_unified, limit 128
 
 # prelude: log to the Prelude Hybrid IDS system
 # ---------------------------------------------
