--- mrtg-old	2003-08-18 12:11:59.000000000 +0200
+++ mrtg	2003-08-19 11:09:53.000000000 +0200
@@ -535,6 +535,7 @@
                          "DS:ds0:$up_abs:$minhb:0:$absi",
                          "DS:ds1:$up_abs:$minhb:0:$abso",
                          "RRA:AVERAGE:0.5:1:$rows",
+                         "RRA:HWPREDICT:1440:0.1:0.0035:288",
                          ( $interval < 30  ? ("RRA:AVERAGE:0.5:".int(30/$interval).":800"):()),
                          "RRA:AVERAGE:0.5:".int(120/$interval).":800",
                          "RRA:AVERAGE:0.5:".int(1440/$interval).":800",
