--- base_qry_alert.orig.php	2006-09-06 11:14:58.000000000 +0200
+++ base_qry_alert.php	2006-10-17 12:10:15.000000000 +0200
@@ -264,6 +264,14 @@
   $encoding = $myrow4[3];
   $detail = $myrow4[4];
 
+  $sensor_name = $myrow4[0];
+  $num="(\*|[0-9]{1,3}|^1?\d\d$|2[0-4]\d|25[0-5])";
+ 
+  if(preg_match("/($num\.$num\.$num\.$num)(.*)/",$sensor_name,$matches)){
+  $sensor_name = gethostbyaddr($matches[1]);
+  $sensor_name = $sensor_name . $matches[6];
+  }      
+
   echo '
        <BLOCKQUOTE>
        <TABLE BORDER=1 width="90%">
@@ -288,7 +296,7 @@
                        <TD class="plfieldhdr">'._INTERFACE.'</TD>
                        <TD class="plfieldhdr">'._FILTER.'</TD>
                   </TR>
-                  <TR><TD class="plfield">'.$myrow4[0].'</TD>
+                  <TR><TD class="plfield">'.$sensor_name.'</TD>
                       <TD class="plfield">'.$myrow4[1].'</TD>
                       <TD class="plfield">'.
                       ( ($myrow4[2] == "") ? "&nbsp;<I>"._NONE."</I>&nbsp;" : $myrow4[2] ).'</TD>
--- base_stat_sensor.orig.php	2006-09-06 11:14:58.000000000 +0200
+++ base_stat_sensor.php	2006-10-17 12:21:51.000000000 +0200
@@ -127,6 +127,8 @@
                 "last_d", "", " ORDER BY last_timestamp DESC");
 
   $sort_sql = $qro->GetSortSQL($qs->GetCurrentSort(), "");
+  $sort_sql[0] = "";
+  $sort_sql[1] = "";
 
   $sql = "SELECT DISTINCT acid_event.sid, count(acid_event.cid) as event_cnt,".
          " count(distinct(acid_event.signature)) as sig_cnt, ".
@@ -154,9 +156,25 @@
   $qro->PrintHeader();
 
   $i = 0;
+  /* Begin sort modification, DK */
+  // 2006-08-04
   while ( ($myrow = $result->baseFetchRow()) && ($i < $qs->GetDisplayRowCnt()) )
   {
+    $temp[$i] = array();
+    $temp[$i]['id'] = $i;
+    $temp[$i]['sensor_id'] = $myrow[0];
     $sensor_id = $myrow[0];
+    $sensor_name = GetSensorName($sensor_id, $db);
+
+    $num="(\*|[0-9]{1,3}|^1?\d\d$|2[0-4]\d|25[0-5])";
+
+    if(preg_match("/($num\.$num\.$num\.$num)(.*)/",$sensor_name,$matches)){
+  	$sensor_name = gethostbyaddr($matches[1]);
+    	$sensor_name = $sensor_name . $matches[6];
+    }
+
+    $temp[$i]['sensor_name'] = $sensor_name;
+
     $event_cnt = $myrow[1];
     $unique_event_cnt = $myrow[2];
     $num_src_ip = $myrow[3];
@@ -165,29 +183,56 @@
     $stop_time = $myrow[6];
 
     /* Print out */ 
-    qroPrintEntryHeader($i);    
 
     $tmp_rowid = $sensor_id;
-    echo '    <TD><INPUT TYPE="checkbox" NAME="action_chk_lst['.$i.']" VALUE="'.$tmp_rowid.'"></TD>';
-    echo '        <INPUT TYPE="hidden" NAME="action_lst['.$i.']" VALUE="'.$tmp_rowid.'">';
+    $temp[$i]['filler1'] = '    <TD><INPUT TYPE="checkbox" NAME="action_chk_lst['.$i.']" VALUE="'.$tmp_rowid.'"></TD>';
+    $temp[$i]['filler2'] = '        <INPUT TYPE="hidden" NAME="action_lst['.$i.']" VALUE="'.$tmp_rowid.'">';
 
-    qroPrintEntry($sensor_id);
-    qroPrintEntry(GetSensorName($sensor_id, $db));
-    qroPrintEntry('<A HREF="base_qry_main.php?new=1&sensor='.$sensor_id.
-                  '&num_result_rows=-1&submit='._QUERYDBP.'">'.
-                  $event_cnt.'</A>');
-
-     qroPrintEntry(BuildUniqueAlertLink("?sensor=".$sensor_id).$unique_event_cnt.'</A>');
-     qroPrintEntry(BuildUniqueAddressLink(1, "&sensor=".$sensor_id).$num_src_ip.'</A>');
-     qroPrintEntry(BuildUniqueAddressLink(2, "&sensor=".$sensor_id).$num_dst_ip.'</A>');
-     qroPrintEntry($start_time);
-     qroPrintEntry($stop_time);
-
-     qroPrintEntryFooter();
+    $temp[$i]['filler3'] = '<A HREF="base_qry_main.php?new=1&sensor='.$sensor_id.  '&num_result_rows=-1&submit='._QUERYDBP.'">'.  $event_cnt.'</A>';
+    $temp[$i]['filler4'] = BuildUniqueAlertLink("?sensor=".$sensor_id).$unique_event_cnt.'</A>';
+    $temp[$i]['filler5'] = BuildUniqueAddressLink(1, "&sensor=".$sensor_id).$num_src_ip.'</A>';
+    $temp[$i]['filler6'] = BuildUniqueAddressLink(2, "&sensor=".$sensor_id).$num_dst_ip.'</A>';
+    $temp[$i]['filler7'] = $start_time;
+    $temp[$i]['filler8'] = $stop_time;
 
      $i++;
   }
 
+  function reorder_sensors_d($a, $b){
+           return ($a["sensor_name"] > $b["sensor_name"]) ? true : false;
+  }
+  function reorder_sensors_a($a, $b){
+           return ($a["sensor_name"] < $b["sensor_name"]) ? true : false;
+  }
+
+
+
+  if(!isset($_GET["sort_order"]) || $_GET["sort_order"] == "sname_d"){
+  usort($temp, "reorder_sensors_d");
+  }elseif($_GET["sort_order"] == "sname_a"){
+  usort($temp, "reorder_sensors_a");
+  }
+  
+ 
+  foreach($temp as $temp_detail){
+    qroPrintEntryHeader($temp_detail['id']);
+    echo $temp_detail['filler1'];
+    echo $temp_detail['filler2'];
+    qroPrintEntry($temp_detail['sensor_id']);
+    $sensor_name = $temp_detail['sensor_name'];
+    list($sensor_name_temp, $extra) = split(":",$sensor_name,2);
+    qroPrintEntry($sensor_name_temp);
+    qroPrintEntry($temp_detail['filler3']);
+    qroPrintEntry($temp_detail['filler4']);
+    qroPrintEntry($temp_detail['filler5']);
+    qroPrintEntry($temp_detail['filler6']);
+    qroPrintEntry($temp_detail['filler7']);
+    qroPrintEntry($temp_detail['filler8']);
+    qroPrintEntryFooter();
+  }
+
+  /* End sort modification, DK */
+
   $result->baseFreeRows();
 
   $qro->PrintFooter();
